docs/highlighting.md

```diff
@@ -0,0 +1,390 @@
+| Raw token data | raw-token-data, raw
+| Rebol | rebol, reb, r, r2, r3
+| Red | red, reds
+| Redcode | redcode, cw
+| RenderScript | renderscript, rs, rsh
+| RobotFramework | robotframework, robot
+| Rouge | rouge, rg
+| Ruby | ruby, rb, builder, fcgi, gemspec, god, irbrc, jbuilder, mspec, pluginspec, podspec, rabl, rake, rbuild, rbw, rbx, ru, thor, watchr
+| Rust | rust, rs
+| SAS | sas
+| SCSS | scss
+| SPARQL | sparql, rq
+| SQF | sqf, hqf
+| SQL | sql, cql, ddl, prc, tab, udf, viw
+| SQLPL | sqlpl, sql, db2
+| STON | ston
+| SVG | svg
+| Sage | sage, sagews
+| SaltStack | saltstack, sls
+| Sass | sass
+| Scala | scala, sbt, sc
+| Scaml | scaml
+| Scheme | scheme, scm, sld, sls, sps, ss
+| Scilab | scilab, sci, sce, tst
+| Self | self
+| Shell | shell, sh, bash, bats, cgi, command, fcgi, ksh, tmux, tool, zsh
+| ShellSession | shellsession, sh-session
+| Shen | shen
+| Slash | slash, sl
+| Slim | slim
+| Smalltalk | smalltalk, st, cs
+| Smarty | smarty, tpl
+| SourcePawn | sourcepawn, sp, sma
+| Squirrel | squirrel, nut
+| Standard ML | standard-ml, ML, fun, sig, sml
+| Stata | stata, do, ado, doh, ihlp, mata, matah, sthlp
+| Stylus | stylus, styl
+| SuperCollider | supercollider, scd, sc
+| Swift | swift
+| SystemVerilog | systemverilog, sv, svh, vh
+| TOML | toml
+| TXL | txl
+| Tcl | tcl, adp, tm
+| Tcsh | tcsh, csh
+| TeX | tex, aux, bbx, bib, cbx, cls, dtx, ins, lbx, ltx, mkii, mkiv, mkvi, sty, toc
+| Tea | tea
+| Text | text, txt, fr
+| Textile | textile
+| Thrift | thrift
+| Turing | turing, t, tu
+| Turtle | turtle, ttl
+| Twig | twig
+| TypeScript | typescript, ts
+| Unified Parallel C | unified-parallel-c, upc
+| UnrealScript | unrealscript, uc
+| VCL | vcl
+| VHDL | vhdl, vhd, vhf, vhi, vho, vhs, vht, vhw
+| Vala | vala, vapi
+| Verilog | verilog, v, veo
+| VimL | viml, vim
+| Visual Basic | visual-basic, vb, bas, cls, frm, frx, vba, vbhtml, vbs
+| Volt | volt
+| Web Ontology Language | web-ontology-language, owl
+| WebIDL | webidl
+| XC | xc
+| XML | xml, ant, axml, ccxml, clixml, cproject, csproj, ct, dita, ditamap, ditaval, dllconfig, filters, fsproj, fxml, glade, grxml, ivy, jelly, kml, launch, mm, mxml, nproj, nuspec, odd, osm, plist, pluginspec, ps1xml, psc1, pt, rdf, rss, scxml, srdf, storyboard, stTheme, sublime-snippet, targets, tmCommand, tml, tmLanguage, tmPreferences, tmSnippet, tmTheme, ts, ui, urdf, vbproj, vcxproj, vxml, wsdl, wsf, wxi, wxl, wxs, x3d, xacro, xaml, xib, xlf, xliff, xmi, xmldist, xsd, xul, zcml
+| XProc | xproc, xpl
+| XQuery | xquery, xq, xql, xqm, xqy
+| XS | xs
+| XSLT | xslt, xsl
+| Xojo | xojo, xojo_code, xojo_menu, xojo_report, xojo_script, xojo_toolbar, xojo_window
+| Xtend | xtend
+| YAML | yaml, yml, reek, rviz
+| Zephir | zephir, zep
+| Zimpl | zimpl, zmpl, zpl
+| desktop | desktop, desktopin
+| eC | ec, eh
+| edn | edn
+| fish | fish
+| mupad | mupad, mu
+| nesC | nesc, nc
+| ooc | ooc
+| reStructuredText | restructuredtext, rst, rest
+| wisp | wisp
+| xBase | xbase, prg
```

Comment:
I [suggested](https://github.com/github/linguist/issues/2278) linking from the GFM help page to a plain list of language codes (instead of a YAML file) that can be used for marking up fenced code blocks. For that purpose I proposed constraining the list to a handful of meaningful values, e.g. `js` for JavaScript (and *maybe* also `javascript`). For that use case I think listing all of these is overkill.